types/src/event.rs

```diff
@@ -89,9 +100,8 @@ impl From<&EventKey> for [u8; EventKey::LENGTH] {
     }
 }
 
-// TODO(#1307)
 impl ser::Serialize for EventKey {
-    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
     where
         S: ser::Serializer,
     {
```

Comment:
should it be `self.to_hex()` like what `AccountAddress` did? as to_string() maybe easily changed by someone who may prefer different format.

Reply:
The exact method used isn't really super important, but I did add a test to ensure that it serializes the whole value properly to json